application/controllers/ReportsController.php

```diff
@@ -127,14 +117,27 @@ public function newAction()
 
         $form = (new ReportForm())
             ->setAction((string) Url::fromRequest())
+            ->setRenderCreateAndShowButton($class !== null)
             ->populate([
                 'filter'    => $this->params->shift('filter'),
                 'reportlet' => $class
             ])
-            ->on(ReportForm::ON_SUCCESS, function () {
-                $this->getResponse()->setHeader('X-Icinga-Container', 'modal-content', true);
-
-                $this->redirectNow('__CLOSE__');
+            ->on(ReportForm::ON_SUCCESS, function (ReportForm $form) use ($class) {
+                Notification::success($this->translate('Created report successfully'));
+
+                $url = Url::fromPath('reporting/reports');
+                if ($form->getPressedSubmitElement()->getName() === 'create_show') {
+                    $url = Url::fromPath('reporting/report', ['id' => $form->getId()]);
+                } elseif ($class !== null) {
+                    $url = '__CLOSE__';
+
+                    // This will close only the modal view and leave the object detail view of Icinga DB Web untouched.
+                    $this
```

Comment:
Don't break here pls, looks weird. You continue calling methods on the response. Not on the controller.